main.c

```diff
@@ -179,6 +180,31 @@ choose_memory_type_index(struct vkcube *vc, uint32_t allowed_memory_types,
    return -1;
 }
 
+static int choose_device(struct vkcube *vc)
+{
+   DIR *dir = opendir("/dev/dri/");
+   if (dir == NULL) {
+      fprintf(stderr, "Failed to open /dev/dri/ directory\n");
+      return -1;
+   }
+
+   struct dirent *entry;
+   while ((entry = readdir(dir)) != NULL) {
+      if (strncmp(entry->d_name, "card", 4) == 0) {
+         char device_path[256];
+	 snprintf(device_path, sizeof(device_path),
+                  "/dev/dri/%s", entry->d_name);
+	 vc->fd = open(device_path, O_RDWR);
+	 if (vc->fd >= 0)
+            return vc->fd;
+      }
+   }
+
+   fprintf(stderr, "failed to open card \n");
+   closedir(dir);
+   return -1;
+}
+
```

Comment:
Please move this commit to a separate MR, because it is independent and unrelated to modifiers. The commit looks nearly ready for merge, btw. I'll merge it asap if it's a separate MR.